doc/api/url.md

```diff
@@ -250,7 +250,470 @@ properties of URL objects:
+
+myURL.protocol = 'ftp';
+console.log(myURL.href);
+  // Prints ftp://example.org
+```
+
+Using the `delete` keyword (e.g. `delete myURL.protocol`) has no effect but will
+still return `true`.
+
+Invalid URL protocol values assigned to the `protocol` property are ignored.
+
+#### url.search
+
+Gets and sets the serialized querystring portion of the URL.
+
+```js
+const myURL = new URL('http://example.org/abc?123');
+console.log(myURL.search);
+  // Prints ?123
+
+myURL.search = 'abc=xyz';
+console.log(myURL.href);
+  // Prints http://example.org/abc?abc=xyz
+```
+
+Using the `delete` keyword (e.g. `delete myURL.search`) has no effect but will
+still return `true`.
+
+Any invalid URL characters appearing in the value assigned the `search`
+property will be percent-encoded. Note that the selection of which characters to
+percent-encode may vary somewhat from what the [`url.parse()`][] and
+[`url.format()`][] methods will produce.
+
+#### url.searchParams
+
+Gets a [`URLSearchParams`][#url_class_urlsearchparams] object representing the
+querystring parameters of the URL.
+
+#### url.username
+
+Gets and sets the username portion of the URL.
+
+```js
+const myURL = new URL('http://abc:xyz@example.com');
+console.log(myURL.username);
+  // Prints abc
+
+myURL.username = '123';
+console.log(myURL.href);
+  // Prints http://123:xyz@example.com
+```
+
+Using the `delete` keyword (e.g. `delete myURL.username`) has no effect but will
+still return `true`.
+
+Any invalid URL characters appearing in the value assigned the `username`
+property will be percent-encoded. Note that the selection of which characters to
+percent-encode may vary somewhat from what the [`url.parse()`][] and
+[`url.format()`][] methods will produce.
+
+#### url.toString([unicode])
+
+* `unicode` {Boolean} If `true`, Unicode characters within the hostname will
+  be encoded as-is without punycode encoding. Defaults to `false`.
+
+The `toString()` method on the `URL` object returns the serialized URL. When
+`unicode` is `false`, the value returned is equivalent to that of `url.href`.
+
+### Class: URLSearchParams
+
+The `URLSearchParams` object provides read and write access to the querystring
+of a `URL`.
+
+```js
+const myURL = new URL('http://example.org/?abc=123');
+console.log(myURL.searchParams.get('abc'));
+  // Prints 123
+
+myURL.searchParams.append('abc', 'xyz');
+console.log(myURL.href);
+  // Prints http://example.org/?abc=123&abc=xyz
+
+myURL.searchParams.delete('abc');
+myURL.searchParams.set('a', 'b');
+console.log(myURL.href);
+  // Prints http://example.org/?a=b
+```
+
+#### Constructor: new URLSearchParams([init])
+
+* `init` {String} The URL querystring
+
+#### urlSearchParams.append(name, value)
+
+* `name` {String}
+* `value` {String}
+
+Append a new name-value pair to the query string.
+
+#### urlSearchParams.delete(name)
+
+* `name` {String}
+
+Remove all name-value pairs whose name is `name`.
+
+#### urlSearchParams.entries()
+
+* Returns: ES6 Iterator
```

Comment:
While at it, why don't you add an [MDN link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#The_iterator_protocol) for the `Iterator` type to [`tools/doc/type-parser.js`](https://github.com/nodejs/node/blob/52c7f9d22115a3ab7174654cc61c9e34335246e2/tools/doc/type-parser.js#L22) ? Then, you will be able to use `{Iterator}` with automatic linking.